src/DataProvider.php

```diff
@@ -41,9 +41,18 @@ public static function getSubscribedEvents(): array
 
     public function onKernelRequest(RequestEvent $event): void
     {
-        if ($event->isMasterRequest()) {
-            $this->httpRequest = $event->getRequest();
-            $this->timestampStart = microtime(true);
+        if (method_exists($event, 'isMainRequest')) { //Symfony >= 5.3
+            if ($event->isMainRequest()) {
+                $this->httpRequest = $event->getRequest();
+                $this->timestampStart = microtime(true);
+            }
+        }
```

Comment:
Nitpicking, but we could return here early. Some Symfony versions have both methods available, and both blocks would be executed.

```suggestion
        return;
        }
```

Reply:
@zoka123 oh yes, this makes very good sense.